cpp/src/arrow/python/python_to_arrow.cc

```diff
@@ -880,7 +881,7 @@ class DecimalConverter
   }
 
  private:
-  const DecimalType* decimal_type_ = nullptr;
+  std::shared_ptr<DecimalType> decimal_type_ = nullptr;
```

Comment:
This change causes the following error with g++ 4.8.5 on CentOS:

```text
/root/rpmbuild/BUILD/apache-arrow-0.15.0/cpp/src/arrow/python/python_to_arrow.cc: In instantiation of 'arrow::Status arrow::py::GetConverterFlat(const std::shared_ptr<arrow::DataType>&, bool, std::unique_ptr<arrow::py::SeqConverter>*) [with arrow::py::NullCoding null_coding = (arrow::py::NullCoding)1]':
/root/rpmbuild/BUILD/apache-arrow-0.15.0/cpp/src/arrow/python/python_to_arrow.cc:1001:5:   required from here
/root/rpmbuild/BUILD/apache-arrow-0.15.0/cpp/src/arrow/python/python_to_arrow.cc:864:7: error: conversion from 'std::nullptr_t' to non-scalar type 'std::shared_ptr<arrow::DecimalType>' requested
 class DecimalConverter
       ^
/root/rpmbuild/BUILD/apache-arrow-0.15.0/cpp/src/arrow/python/python_to_arrow.cc:894:10: note: synthesized method 'arrow::py::DecimalConverter<(arrow::py::NullCoding)1>::DecimalConverter()' first required here 
     *out = std::unique_ptr<SeqConverter>(new TYPE_CLASS<null_coding>); \
          ^
/root/rpmbuild/BUILD/apache-arrow-0.15.0/cpp/src/arrow/python/python_to_arrow.cc:915:5: note: in expansion of macro 'SIMPLE_CONVERTER_CASE'
     SIMPLE_CONVERTER_CASE(DECIMAL, DecimalConverter);
     ^
/root/rpmbuild/BUILD/apache-arrow-0.15.0/cpp/src/arrow/python/python_to_arrow.cc: In instantiation of 'arrow::Status arrow::py::GetConverterFlat(const std::shared_ptr<arrow::DataType>&, bool, std::unique_ptr<arrow::py::SeqConverter>*) [with arrow::py::NullCoding null_coding = (arrow::py::NullCoding)0]':
/root/rpmbuild/BUILD/apache-arrow-0.15.0/cpp/src/arrow/python/python_to_arrow.cc:1004:5:   required from here
/root/rpmbuild/BUILD/apache-arrow-0.15.0/cpp/src/arrow/python/python_to_arrow.cc:864:7: error: conversion from 'std::nullptr_t' to non-scalar type 'std::shared_ptr<arrow::DecimalType>' requested
 class DecimalConverter
       ^
/root/rpmbuild/BUILD/apache-arrow-0.15.0/cpp/src/arrow/python/python_to_arrow.cc:894:10: note: synthesized method 'arrow::py::DecimalConverter<(arrow::py::NullCoding)0>::DecimalConverter()' first required here 
     *out = std::unique_ptr<SeqConverter>(new TYPE_CLASS<null_coding>); \
          ^
/root/rpmbuild/BUILD/apache-arrow-0.15.0/cpp/src/arrow/python/python_to_arrow.cc:915:5: note: in expansion of macro 'SIMPLE_CONVERTER_CASE'
     SIMPLE_CONVERTER_CASE(DECIMAL, DecimalConverter);
     ^
```

I opened a pull request to fix this: https://github.com/apache/arrow/pull/5456